zonemaster/zonemaster
# Centralized documentation

## Purpose

This PR centralizes the documentation, putting together documentation files from across components into the main zonemaster/zonemaster repository. This will help browsing the documentation and allow at a later stage to publish it.

## Context

Discussions at F2F 2022-06 and 2023-01.

## Changes

Copy some documentation files to zonemaster/zonemaster (see commits).

<details><summary>New tree structure of docs/</summary>

```
docs/
├── contact-and-mailing-lists.md
├── internal
│   ├── design
│   │   ├── Architecture Notes.md
│   │   ├── Implementation Guidelines.md
│   │   ├── Profiles.md
│   │   ├── README.md
│   │   └── Versions and Releases.md
│   ├── distrib-testing
│   │   ├── BuildEnvironmentPreparation.md
│   │   ├── CentOS-build-environment.md
│   │   ├── Debian-build-environment.md
│   │   ├── FreeBSD-build-environment.md
│   │   ├── Ubuntu-build-environment.md
│   │   └── Ubuntu-Node.js-build-environment.md
│   ├── functional-tests
│   │   ├── Behavior-TP
│   │   │   ├── behavior01.md
│   │   │   ├── behavior02.md
│   │   │   ├── behavior03.md
│   │   │   ├── behavior04.md
│   │   │   ├── behavior05.md
│   │   │   ├── behavior06.md
│   │   │   ├── behavior07.md
│   │   │   ├── behavior08.md
│   │   │   └── behavior09.md
│   │   ├── Configuration-TP
│   │   │   ├── configuration01.md
│   │   │   ├── configuration02.md
│   │   │   ├── configuration03.md
│   │   │   └── configuration04.md
│   │   └── Restriction-TP
│   │       ├── restriction01.md
│   │       ├── restriction02.md
│   │       └── restriction03.md
│   ├── maintenance
│   │   ├── ChangeManagement.md
│   │   ├── Instructions-for-translators.md
│   │   ├── IssueLabels.md
│   │   ├── ReleaseProcess-create-docker-image.md
│   │   ├── ReleaseProcess-create-test-distribution.md
│   │   ├── ReleaseProcess-deb-packages.md
│   │   ├── ReleaseProcess.md
│   │   ├── ReleaseProcess-preparation.md
│   │   ├── ReleaseProcess-release.md
│   │   ├── ReleaseProcess-test-planning.md
│   │   ├── StyleGuide.md
│   │   ├── SupportCriteria.md
│   │   └── SystemTesting.md
│   ├── requirements
│   │   ├── CLI-Combined-Requirements.md
│   │   ├── DefineTests.md
│   │   ├── DNSCheck-Batch-Requirements.txt
│   │   ├── DNSCheck-CLI-Requirements.txt
│   │   ├── DNSCheck-GUI-Requirements.txt
│   │   ├── Engine-Functional-Test-Requirements.md
│   │   ├── ExistingZCFeaturesGUI.md
│   │   ├── ExistingZCFeatures.md
│   │   ├── FunctionalRequirements.md
│   │   ├── GUI-Combined-Requirements.txt
│   │   ├── GUI-Functional-Test-Requirements.md
│   │   └── README.md
│   ├── templates
│   │   └── specifications
│   │       └── tests
│   │           ├── MessageTagSpecification.md
│   │           ├── Template01.md
│   │           └── TestCaseIdentifierSpecification.md
│   └── test-requirements
│       ├── README.md
│       ├── TestRequirements.md
│       └── TestRequirements-table-specification.txt
└── public
    ├── book.toml
    ├── configuration
    │   ├── backend-environment-variables.md
    │   ├── backend.md
    │   ├── gui.md
    │   └── profiles.md
    ├── installation
    │   ├── docker.md
    │   ├── prerequisites.md
    │   ├── zonemaster-backend.md
    │   ├── zonemaster-cli.md
    │   ├── zonemaster-engine.md
    │   ├── zonemaster-gui.md
    │   └── zonemaster-ldns.md
    ├── specifications
    │   ├── README.md
    │   ├── tests
    │   │   ├── Address-TP
    │   │   │   ├── address01.md
    │   │   │   ├── address02.md
    │   │   │   ├── address03.md
    │   │   │   └── README.md
    │   │   ├── Basic-TP
    │   │   │   ├── basic00.md
    │   │   │   ├── basic01.md
    │   │   │   ├── basic02.md
    │   │   │   ├── basic03.md
    │   │   │   └── README.md
    │   │   ├── Connectivity-TP
    │   │   │   ├── connectivity01.md
    │   │   │   ├── connectivity02.md
    │   │   │   ├── connectivity03.md
    │   │   │   └── README.md
    │   │   ├── Consistency-TP
    │   │   │   ├── consistency01.md
    │   │   │   ├── consistency02.md
    │   │   │   ├── consistency03.md
    │   │   │   ├── consistency04.md
    │   │   │   ├── consistency05.md
    │   │   │   ├── consistency06.md
    │   │   │   └── README.md
    │   │   ├── Delegation-TP
    │   │   │   ├── delegation01.md
    │   │   │   ├── delegation02.md
    │   │   │   ├── delegation03.md
    │   │   │   ├── delegation04.md
    │   │   │   ├── delegation05.md
    │   │   │   ├── delegation06.md
    │   │   │   ├── delegation07.md
    │   │   │   └── README.md
    │   │   ├── DNSQueryAndResponseDefaults.md
    │   │   ├── DNSSEC-TP
    │   │   │   ├── dnssec01.md
    │   │   │   ├── dnssec02.md
    │   │   │   ├── dnssec03.md
    │   │   │   ├── dnssec04.md
    │   │   │   ├── dnssec05.md
    │   │   │   ├── dnssec06.md
    │   │   │   ├── dnssec07.md
    │   │   │   ├── dnssec08.md
    │   │   │   ├── dnssec09.md
    │   │   │   ├── dnssec10.md
    │   │   │   ├── dnssec11.md
    │   │   │   ├── dnssec12.md
    │   │   │   ├── dnssec13.md
    │   │   │   ├── dnssec14.md
    │   │   │   ├── dnssec15.md
    │   │   │   ├── dnssec16.md
    │   │   │   ├── dnssec17.md
    │   │   │   ├── dnssec18.md
    │   │   │   └── README.md
    │   │   ├── ImplementedTestCases.md
    │   │   ├── MasterTestPlan.md
    │   │   ├── Methods.md
    │   │   ├── MethodsV2.md
    │   │   ├── Nameserver-TP
    │   │   │   ├── nameserver01.md
    │   │   │   ├── nameserver02.md
    │   │   │   ├── nameserver03.md
    │   │   │   ├── nameserver04.md
    │   │   │   ├── nameserver05.md
    │   │   │   ├── nameserver06.md
    │   │   │   ├── nameserver07.md
    │   │   │   ├── nameserver08.md
    │   │   │   ├── nameserver09.md
    │   │   │   ├── nameserver10.md
    │   │   │   ├── nameserver11.md
    │   │   │   ├── nameserver12.md
    │   │   │   ├── nameserver13.md
    │   │   │   ├── nameserver14.md
    │   │   │   └── README.md
    │   │   ├── README.md
    │   │   ├── RequirementsAndNormalizationOfDomainNames.md
    │   │   ├── SeverityLevelDefinitions.md
    │   │   ├── Syntax-TP
    │   │   │   ├── README.md
    │   │   │   ├── syntax01.md
    │   │   │   ├── syntax02.md
    │   │   │   ├── syntax03.md
    │   │   │   ├── syntax04.md
    │   │   │   ├── syntax05.md
    │   │   │   ├── syntax06.md
    │   │   │   ├── syntax07.md
    │   │   │   └── syntax08.md
    │   │   ├── TestMessages.md
    │   │   └── Zone-TP
    │   │       ├── README.md
    │   │       ├── zone01.md
    │   │       ├── zone02.md
    │   │       ├── zone03.md
    │   │       ├── zone04.md
    │   │       ├── zone05.md
    │   │       ├── zone06.md
    │   │       ├── zone07.md
    │   │       ├── zone08.md
    │   │       ├── zone09.md
    │   │       └── zone10.md
    │   └── test-types
    │       ├── README.md
    │       └── undelegated-test.md
    ├── SUMMARY.md
    ├── upgrade
    │   └── backend.md
    └── using
        ├── backend
        │   ├── getting-started.md
        │   ├── README.md
        │   ├── rpcapi-reference.md
        │   └── telemetry.md
        ├── cli.md
        └── gui
            ├── api.md
            └── README.md
```
</details>

## How to test this PR

* Review the new structure
* Check for missing documents
* Check for unnecessary documents


## Further work

This is the first step of a bigger work that wants to centralize the Zonemaster documentation to then publish it on its own using [mdbook](https://rust-lang.github.io/mdBook/).

1. Copy documents from other repositories to zonemaster/zonemaster in a comprehensive structure (this PR)
2. Remove documents from other repositories to keep only the file in zonemaster/zonemaster
3. Review documentation and improve if needed
4. Publish it using [mdbook](https://rust-lang.github.io/mdBook/)
    ```
    cd docs/public
    mdbook build
    open book/index.html
    ```
